torchtune/utils/multi_dataset.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Tuple
+
+from omegaconf import DictConfig
+from torch.utils.data import Dataset
+from torchtune import config, utils
+from tqdm import tqdm
+
+log = utils.get_logger("DEBUG")
+
+
+class MultiDataset(Dataset):
+    def __init__(self, datasets: DictConfig[DictConfig], tokenizer):
```

Comment:
I'm not sure about passing in the datasets as config objects, because then this class is hard to use without having to create a config. And if you ever want to play around with `MultiDataset` in a notebook environment, this might get tedious. Config objects should be separate from the actual code as much as possible, except in the recipe where components need to be instantiated. But I see that it's hard to create a `MultiDataset` without having to instantiate the individual datasets first...

What do you think about an alternate approach where we instantiate the datasets outside of this class and then pass in the dataset objects?
```
# in recipe
datasets = [config.instantiate(single_cfg_dataset, tokenizer) for single_cfg_dataset in cfg_dataset]
ds = MultiDataset(datasets=datasets)

# Modified MultiDataset constructor - won't need tokenizer anymore
def __init__(self, datasets: List[Dataset]):
```

Reply:
Clever idea, patch added.